### PR TITLE
[stackdriver-exporter] feat: Adding relabeling config to serviceMonitor

### PR DESCRIPTION
**Pull request by cleberwarmling**
Adding relabeling configs to its serviceMonitor. Will be useful to improve Prometheus cardinality.

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Comment by luizm** [comment]
LGTM :)

**Review by caarlos0 — DISMISSED** [review]

File: charts/prometheus-stackdriver-exporter/templates/servicemonitor.yaml [hunk 1]
```diff
@@ -22,6 +22,14 @@ spec:
     scrapeTimeout: {{ .Values.serviceMonitor.scrapeTimeout }}
     {{- end }}
     honorLabels: {{ .Values.serviceMonitor.honorLabels }}
+    {{- if .Values.serviceMonitor.metricRelabelings }}
+    metricRelabelings:
+{{ toYaml .Values.serviceMonitor.metricRelabelings | indent 4 }}
```
Comment (torstenwalter) [inline charts/prometheus-stackdriver-exporter/templates/servicemonitor.yaml:27]:
```suggestion
    {{ toYaml .Values.serviceMonitor.metricRelabelings | nindent 4 }}
```
Could you check if that also works?

File: charts/prometheus-stackdriver-exporter/templates/servicemonitor.yaml [hunk 2]
```diff
@@ -22,6 +22,14 @@ spec:
...
+    {{- end }}
+    {{- if .Values.serviceMonitor.relabelings }}
+    relabelings:
+{{ toYaml .Values.serviceMonitor.relabelings | indent 4 }}
```
Comment (torstenwalter) [inline charts/prometheus-stackdriver-exporter/templates/servicemonitor.yaml:31]:
```suggestion
    {{ toYaml .Values.serviceMonitor.relabelings | nindent 4 }}
```

File: charts/prometheus-stackdriver-exporter/templates/servicemonitor.yaml (hunk 1 above)
Comment (cleberwarmling) [reply charts/prometheus-stackdriver-exporter/templates/servicemonitor.yaml:27]: yeap, it works, also trimming whitespaces gives a cleaner template output... But then also there are a dozen of those occurrences in the Chart. WDYT, it worth changing those also?

Comment (torstenwalter) [reply charts/prometheus-stackdriver-exporter/templates/servicemonitor.yaml:27]: Sure maybe in a separate PR?

Comment (cleberwarmling) [reply charts/prometheus-stackdriver-exporter/templates/servicemonitor.yaml]: @torstenwalter changed on this PR. Seems to be just it.

**Review by torstenwalter — APPROVED** [review]

**Review by monotek — APPROVED** [review]

**Comment by cleberwarmling** [comment]
@apenney @rpahli

**Review by rpahli — APPROVED** [review]
LGTM :-)

**Pull request merged**